model/bootstrap/node_info.go

```diff
@@ -219,42 +269,15 @@ type NodePrivateKeys struct {
 	NetworkKey crypto.PrivateKey
 }
 
-// NodeInfo contains information for a node. This is used during the bootstrapping
-// process to represent each node. When writing node information to disk, use
-// `Public` or `Private` to obtain the appropriate canonical structure.
-//
-// A NodeInfo instance can contain EITHER public keys OR private keys, not both.
-// This can be ensured by using only using the provided constructors and NOT
-// manually constructing an instance.
-type NodeInfo struct {
-
-	// NodeID is the unique identifier of the node in the network
-	NodeID flow.Identifier
-
-	// Role is the flow role of the node (collection, consensus, etc...)
-	Role flow.Role
-
-	// Address is the networking address of the node (IP:PORT), not to be
-	// confused with the address of the flow account associated with the node's
-	// machine account.
-	Address string
-
-	// Weight is the weight of the node
-	Weight uint64
-
-	// PRIVATE Variant:
-	networkPrivKey crypto.PrivateKey
-	stakingPrivKey crypto.PrivateKey
-
-	// By convention, `NodeInfo` must either include the public fields and exclude the private fields, or
-	// vice versa. Mixtures are not allowed. Please check function [NodeInfoType] for the precise convention.
-	//
-	// PUBLIC Variant:
-	networkPubKey crypto.PublicKey
-	stakingPubKey crypto.PublicKey
-	stakingPoP    crypto.Signature
+type NodeInfo interface {
+	NodeID() flow.Identifier
+	Identity() *flow.Identity
```

Comment:
I feel this should return `IdentitySkeleton`. The full `Identity` includes the [life cycle status of a node](https://github.com/onflow/flow-go/blob/b57e949c5c84f74d1c97bd81a4c3b9134978dda7/model/flow/identity.go#L117), which is determined by the protocol state on the fly while the protocol runs.

The node identities supplied to the protocol state typically don't contain this cycle status - instead they are [IdentitySkeleton](https://github.com/onflow/flow-go/blob/b57e949c5c84f74d1c97bd81a4c3b9134978dda7/model/flow/identity.go#L116). For example, the [EpochSetup](https://github.com/onflow/flow-go/blob/5a9a5ae1862f46709d524645c3ff0e39cb4d530b/model/flow/epoch.go#L146) only uses `IdentitySkeleton`s.
```suggestion
	Identity() *flow.IdentitySkeleton
```